src/core/api/transform/openrouter-stream.ts

```diff
@@ -140,32 +93,18 @@ export async function createOpenRouterStream(
 	}
 
 	let reasoning: { max_tokens: number } | undefined
-	switch (model.id) {
-		case "anthropic/claude-haiku-4.5":
-		case "anthropic/claude-4.5-haiku":
-		case "anthropic/claude-sonnet-4.5":
-		case "anthropic/claude-4.5-sonnet":
-		case "anthropic/claude-sonnet-4":
-		case "anthropic/claude-opus-4.1":
-		case "anthropic/claude-opus-4":
-		case "anthropic/claude-3.7-sonnet":
-		case "anthropic/claude-3.7-sonnet:beta":
-		case "anthropic/claude-3.7-sonnet:thinking":
-		case "anthropic/claude-3-7-sonnet":
-		case "anthropic/claude-3-7-sonnet:beta":
-			const budget_tokens = thinkingBudgetTokens || 0
-			const reasoningOn = budget_tokens !== 0
-			if (reasoningOn) {
-				temperature = undefined // extended thinking does not support non-1 temperature
-				reasoning = { max_tokens: budget_tokens }
-			}
-			break
-		default:
-			if (thinkingBudgetTokens && model.info?.thinkingConfig && thinkingBudgetTokens > 0) {
-				temperature = undefined // extended thinking does not support non-1 temperature
-				reasoning = { max_tokens: thinkingBudgetTokens }
-				break
-			}
+	if (isAnthropicModel) {
+		const budget_tokens = thinkingBudgetTokens || 0
+		const reasoningOn = budget_tokens !== 0
+		if (reasoningOn) {
+			temperature = undefined // extended thinking does not support non-1 temperature
+			reasoning = { max_tokens: budget_tokens }
+		}
+	} else {
+		if (thinkingBudgetTokens && model.info?.thinkingConfig && thinkingBudgetTokens > 0) {
+			temperature = undefined // extended thinking does not support non-1 temperature
+			reasoning = { max_tokens: thinkingBudgetTokens }
+		}
```

Comment:
Seems like we are setting reasoning to the same value for both anthropic and non-anthropic models?

```suggestion
	if (
		thinkingBudgetTokens &&
		thinkingBudgetTokens > 0 &&
		(isAnthropicModel || Object.keys(model.info?.thinkingConfig || {}).length > 0)
	) {
		temperature = undefined // extended thinking does not support non-1 temperature
		reasoning = { max_tokens: thinkingBudgetTokens }
	}
```